terraform/modules/ingress/certificates.tf

```diff
@@ -0,0 +1,179 @@
+#------------------------------------------------------------------------------
+# Certificate Authority
+#------------------------------------------------------------------------------
+resource "tls_private_key" "root_ingress" {
+  count       = var.tls ? 1 : 0
+  algorithm   = "RSA"
+  ecdsa_curve = "P384"
+  rsa_bits    = "4096"
+}
+
+resource "tls_self_signed_cert" "root_ingress" {
+  count                 = length(tls_private_key.root_ingress)
+  private_key_pem       = tls_private_key.root_ingress.0.private_key_pem
+  is_ca_certificate     = true
+  validity_period_hours = "168"
+  allowed_uses = [
+    "cert_signing",
+    "key_encipherment",
+    "digital_signature"
+  ]
+  subject {
+    organization = "ArmoniK Ingress Root (NonTrusted)"
+    common_name  = "ArmoniK Ingress Root (NonTrusted) Private Certificate Authority"
+    country      = "France"
+  }
+}
+
+#------------------------------------------------------------------------------
+# Client Certificate Authority
+#------------------------------------------------------------------------------
+resource "tls_private_key" "client_root_ingress" {
+  count       = var.mtls ? 1 : 0
+  algorithm   = "RSA"
+  ecdsa_curve = "P384"
+  rsa_bits    = "4096"
+}
+
+resource "tls_self_signed_cert" "client_root_ingress" {
+  count                 = length(tls_private_key.client_root_ingress)
+  private_key_pem       = tls_private_key.client_root_ingress.0.private_key_pem
+  is_ca_certificate     = true
+  validity_period_hours = "168"
+  allowed_uses = [
+    "cert_signing",
+    "key_encipherment",
+    "digital_signature"
+  ]
+  subject {
+    organization = "ArmoniK Client Ingress Root (NonTrusted)"
+    common_name  = "ArmoniK Client Ingress Root (NonTrusted) Private Certificate Authority"
+    country      = "France"
+  }
+}
+
+#------------------------------------------------------------------------------
+# Server Certificate
+#------------------------------------------------------------------------------
+resource "tls_private_key" "ingress_private_key" {
+  count       = length(tls_private_key.root_ingress)
+  algorithm   = "RSA"
+  ecdsa_curve = "P384"
+  rsa_bits    = "4096"
+}
+
+resource "tls_cert_request" "ingress_cert_request" {
+  count           = length(tls_private_key.ingress_private_key)
+  private_key_pem = tls_private_key.ingress_private_key.0.private_key_pem
+  subject {
+    country     = "France"
+    common_name = "localhost"
```

Comment:
`localhost` or `armonik.local` or `ingress.armonik`? I don't really care either way.

Reply:
For SSL validation it's important, so unless I modify the hosts file, localhost is mandatory